airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.test.tsx

```diff
@@ -20,7 +20,9 @@ jest.mock("../useConnectionHook", () => ({
     getConnection: () => mockConnection,
   }),
   useUpdateConnection: () => ({
-    mutateAsync: jest.fn(async (connection: WebBackendConnectionUpdate) => connection),
+    mutateAsync: jest.fn(async (connection: WebBackendConnectionUpdate) => {
+      return { ...mockConnection, ...connection };
+    }),
```

Comment:
This makes the mocked hook a patch-style update rather than a full replace, so it more closely matches the behavior of the backend update API